src/tree/assess/scoreAggregators/ReadabilityScoreAggregator.js

```diff
@@ -0,0 +1,167 @@
+		 * we also check whether papers have text to analyze among other things.
+		 *
+		 * Since only applicable assessments are applied, we only get the results
+		 * of applicable assessments either way. So this check suffices.
+		 */
+		return results.length === totalNrOfAssessments;
+	}
+
+	/**
+	 * Calculates the total score (GOOD, OKAY or NEEDS IMPROVEMENT)
+	 * based on the total penalty score.
+	 *
+	 * @param {boolean} allAssessmentsSupported Whether all assessments are supported for this language.
+	 * @param {number}  penalty                 The total penalty.
+	 *
+	 * @returns {number} The total score.
+	 */
+	calculateScore( allAssessmentsSupported, penalty ) {
+		if ( allAssessmentsSupported ) {
+			if ( penalty > 6 ) {
+				return READABILITY_SCORES.NEEDS_IMPROVEMENT;
+			}
+
+			if ( penalty > 4 ) {
+				return READABILITY_SCORES.OKAY;
+			}
+		} else {
+			if ( penalty > 4 ) {
+				return READABILITY_SCORES.NEEDS_IMPROVEMENT;
+			}
+
+			if ( penalty > 2 ) {
+				return READABILITY_SCORES.OKAY;
+			}
+		}
+		return READABILITY_SCORES.GOOD;
+	}
+
+	/**
+	 * Calculates the total amount of penalty points based on the given assessment results.
+	 *
+	 * @param {AssessmentResult[]} results The valid results from which to calculate the total penalty points.
+	 *
+	 * @returns {number} The total penalty points for the results.
+	 */
+	calculatePenalty( results ) {
+		const penaltyPoints = results.map( result => {
+			// Compute the rating ("error", "feedback", "bad", "ok" or "good").
+			const rating = scoreToRating( result.getScore() );
+
+			if ( this.allAssessmentsSupported( results ) ) {
+				// Default to 0 on "error" or "feedback" or any other not-supported score.
+				return penaltyMappingFullSupport[ rating ] || 0;
+			}
+			return penaltyMappingPartialSupport[ rating ] || 0;
+		} );
+		return sum( penaltyPoints );
+	}
+
+	/**
+	 * Returns the list of valid results.
+	 * Valid results are all results that have a score and a text.
+	 *
+	 * @param {AssessmentResult[]} results The results to filter the valid results from.
+	 *
+	 * @returns {AssessmentResult[]} The list of valid results.
+	 */
+	getValidResults( results ) {
+		return results.filter( result => result.hasScore() && result.hasText() );
+	}
+
+	/**
+	 * Aggregates the given assessment results into a single analysis score.
+	 *
+	 * @param {AssessmentResult[]} results The assessment results.
+	 *
+	 * @returns {number} The aggregated score.
+	 */
+	aggregate( results ) {
+		const validResults = this.getValidResults( results );
+
+		/*
+		 * If you have no content, you have a red indicator.
+		 * (Assume that one result always means the 'no content' assessment result, (see `contentAssessor`) ).
+		 */
+		if ( validResults.length <= 1 ) {
+			return READABILITY_SCORES.NEEDS_IMPROVEMENT;
+		}
+
+		const penalty = this.calculatePenalty( results );
+		return this.calculateScore( this.allAssessmentsSupported( results ), penalty );
```

Comment:
Saved the result of `isFullySupported` to a variable, which I now give as an argument to the function.

The `contentAssessor` checks all applicable assessments. It does not check for validity, since it does not check the assessment results. Either way, we need to change the way language support is checked, since it does not work the way it says it does now.
An assessment can be applicable because of a lot of different reasons, not only because the language is supported.

We could use this function:
```js
const isLanguageAvailable = getLanguageAvailability( paper.getLocale(), availableLanguages );
```
Although we then need a way to get the locale of the paper within the `Aggregator`.